cmd/state-installer/test/integration/installer_int_test.go

```diff
@@ -221,6 +221,50 @@ func (suite *InstallerIntegrationTestSuite) TestInstallerOverwriteServiceApp() {
 	cp.ExpectExit() // the return code can vary depending on shell (e.g. zsh vs. bash); just assert the installer shell exited
 }
 
+func (suite *InstallerIntegrationTestSuite) TestInstallWhileInUse() {
+	suite.OnlyRunForTags(tagsuite.Installer)
+	if runtime.GOOS != "windows" {
+		suite.T().Skip("Only windows can have issues with copying over files in use")
+	}
+
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	dir, err := ioutil.TempDir("", "system*")
+	suite.NoError(err)
+
+	cp := ts.SpawnCmdWithOpts(
+		suite.installerExe,
+		e2e.OptArgs(installationDir(ts)),
+		e2e.OptAppendEnv(constants.DisableUpdates+"=true"),
+		e2e.OptAppendEnv(fmt.Sprintf("%s=%s", constants.OverwriteDefaultSystemPathEnvVarName, dir)),
+	)
+	cp.Expect("successfully installed", e2e.RuntimeSourcingTimeoutOpt)
+	cp.ExpectInput()
+	cp.SendLine("state checkout ActiveState/Perl-5.32")
+	cp.Expect("Checked out")
+	cp.SendLine("state shell Perl-5.32")
+	cp.Expect("Activated") // state.exe remains active
+
+	// On Windows we cannot delete files/executables in use. Instead, the installer copies new
+	// executables into the target directory with the ".new" suffix and renames them to the target
+	// executables. Verify that this works without error.
+	cp2 := ts.SpawnCmdWithOpts(
+		suite.installerExe,
+		e2e.OptArgs(installationDir(ts), "-f"),
+		e2e.OptAppendEnv(constants.DisableUpdates+"=true"),
+		e2e.OptAppendEnv(fmt.Sprintf("%s=%s", constants.OverwriteDefaultSystemPathEnvVarName, dir)),
+	)
+	cp2.Expect("successfully installed", e2e.RuntimeSourcingTimeoutOpt)
+	cp2.ExpectInput()
```

Comment:
If it's easy, an added expect here could be that the `state.old.exe` binary exists.